x/evm/testutil/suite.go

```diff
@@ -0,0 +1,423 @@
+			Block: version.BlockProtocol,
+		},
+		LastBlockId: tmproto.BlockID{
+			Hash: tmhash.Sum([]byte("block_id")),
+			PartSetHeader: tmproto.PartSetHeader{
+				Total: 11,
+				Hash:  tmhash.Sum([]byte("partset_header")),
+			},
+		},
+		AppHash:            tmhash.Sum([]byte("app")),
+		DataHash:           tmhash.Sum([]byte("data")),
+		EvidenceHash:       tmhash.Sum([]byte("evidence")),
+		ValidatorsHash:     tmhash.Sum([]byte("validators")),
+		NextValidatorsHash: tmhash.Sum([]byte("next_validators")),
+		ConsensusHash:      tmhash.Sum([]byte("consensus")),
+		LastResultsHash:    tmhash.Sum([]byte("last_result")),
+	})
+
+	queryHelper := baseapp.NewQueryServerTestHelper(suite.Ctx, suite.App.InterfaceRegistry())
+	types.RegisterQueryServer(queryHelper, suite.App.EvmKeeper)
+	suite.QueryClient = types.NewQueryClient(queryHelper)
+
+	acc := &ethermint.EthAccount{
+		BaseAccount: authtypes.NewBaseAccount(sdk.AccAddress(suite.Address.Bytes()), nil, 0, 0),
+		CodeHash:    common.BytesToHash(crypto.Keccak256(nil)).String(),
+	}
+
+	suite.App.AccountKeeper.SetAccount(suite.Ctx, acc)
+
+	valAddr := sdk.ValAddress(suite.Address.Bytes())
+	validator, err := stakingtypes.NewValidator(valAddr, priv.PubKey(), stakingtypes.Description{})
+	require.NoError(t, err)
+	err = suite.App.StakingKeeper.SetValidatorByConsAddr(suite.Ctx, validator)
+	require.NoError(t, err)
+	err = suite.App.StakingKeeper.SetValidatorByConsAddr(suite.Ctx, validator)
+	require.NoError(t, err)
+	suite.App.StakingKeeper.SetValidator(suite.Ctx, validator)
+
+	encodingConfig := encoding.MakeConfig(app.ModuleBasics)
+	suite.ClientCtx = client.Context{}.WithTxConfig(encodingConfig.TxConfig)
+	suite.EthSigner = ethtypes.LatestSignerForChainID(suite.App.EvmKeeper.ChainID())
+	suite.appCodec = encodingConfig.Codec
+	suite.denom = evmtypes.DefaultEVMDenom
+}
+
+func (suite *KeeperTestSuite) EvmDenom() string {
+	ctx := sdk.WrapSDKContext(suite.Ctx)
+	rsp, _ := suite.QueryClient.Params(ctx, &types.QueryParamsRequest{})
+	return rsp.Params.EvmDenom
+}
+
+// Commit and begin new block
+func (suite *KeeperTestSuite) Commit() {
+	_ = suite.App.Commit()
+	header := suite.Ctx.BlockHeader()
+	header.Height += 1
+	suite.App.BeginBlock(abci.RequestBeginBlock{
+		Header: header,
+	})
+
+	// update ctx
+	suite.Ctx = suite.App.BaseApp.NewContext(false, header)
+
+	queryHelper := baseapp.NewQueryServerTestHelper(suite.Ctx, suite.App.InterfaceRegistry())
+	types.RegisterQueryServer(queryHelper, suite.App.EvmKeeper)
+	suite.QueryClient = types.NewQueryClient(queryHelper)
+}
+
+func (suite *KeeperTestSuite) StateDB() vm.StateDB {
+	return statedb.New(
+		suite.Ctx,
+		suite.App.EvmKeeper,
+		types.NewEmptyTxConfig(common.BytesToHash(suite.Ctx.HeaderHash().Bytes())),
+	)
+}
+
+// DeployTestContract deploy a test erc20 contract and returns the contract address
+func (suite *KeeperTestSuite) DeployTestContract(t require.TestingT, owner common.Address, supply *big.Int) common.Address {
```

Comment:
Is it used anywhere?
Same question for `TransferERC20Token`, `DeployTestMessageCall` methods?

Reply:
ah right the testutil package KeeperTestSuite was copied from the suite in Keeper so they were unused -- now migrated the tests in keeper package to use the shared Suite in testutil so they are no longer unused and no more duplicate suite